### PR TITLE
use SHermitianCompact for symmetric tensors

### PR DESCRIPTION
**Pull request by simonbyrne**
It turns out this was already in StaticArrays

### CODE REVIEW

**Comment by simonbyrne** [comment]
bors try

File: src/Atmos/Model/turbulence.jl [hunk 1]
```diff
@@ -108,24 +107,31 @@ eprint = {https://onlinelibrary.wiley.com/doi/pdf/10.1111/j.2153-3490.1962.tb001
...
   T = eltype(diffusive)
   N² = inv(aux.moisture.θ_v * diffusive.turbulence.∂θ∂Φ)
-  normS = sqrt(2*(S[1]^2 + S[2]^2 + S[3]^2 + 2*(S[4]^2 + S[5]^2 + S[6]^2)))
   Richardson = N² / (normS^2 + eps(normS))
-  buoyancy_factor = N² <= T(0) ? T(1) : sqrt(max(T(0), T(1) - Richardson*inv_Pr_turb))^(T(1//4))
-  return buoyancy_factor
+  if N² <= T(0)
+    T(1) 
+  else  
+    max(T(0), sqrt(T(1) - Richardson*inv_Pr_turb))
```
Comment (asraero) [inline src/Atmos/Model/turbulence.jl]: `sqrt(max(T(0), T(1) - Richardson * inv_Pr_turb))`
(`Richardson` can be large if shear contribution is low)

File: src/Atmos/Model/turbulence.jl [hunk 2]
```diff
@@ -108,24 +107,31 @@ eprint = {https://onlinelibrary.wiley.com/doi/pdf/10.1111/j.2153-3490.1962.tb001
...
+  end 
 end
+
+function strain_rate_magnitude(S::SHermitianCompact)
+  sqrt(2*S[1,1]^2 + 4*S[2,1]^2 + 4*S[3,1]^2 + 2*S[2,2]^2 + 4*S[4,2]^2 + 2*S[3,3])
```
Comment (asraero) [inline src/Atmos/Model/turbulence.jl]:
```suggestion
  sqrt(2*S[1,1]^2 + 4*S[2,1]^2 + 4*S[3,1]^2 + 2*S[2,2]^2 + 4*S[3,2]^2 + 2*S[3,3])
```

**Comment by asraero** [comment]
bors try

File: src/Atmos/Model/turbulence.jl [hunk 3]
```diff
@@ -108,24 +107,27 @@ eprint = {https://onlinelibrary.wiley.com/doi/pdf/10.1111/j.2153-3490.1962.tb001
...
 """
-function buoyancy_correction(S, diffusive::Vars, aux::Vars)
+function squared_buoyancy_correction(normS, diffusive::Vars, aux::Vars)
   T = eltype(diffusive)
   N² = inv(aux.moisture.θ_v * diffusive.turbulence.∂θ∂Φ)
-  normS = sqrt(2*(S[1]^2 + S[2]^2 + S[3]^2 + 2*(S[4]^2 + S[5]^2 + S[6]^2)))
   Richardson = N² / (normS^2 + eps(normS))
-  buoyancy_factor = N² <= T(0) ? T(1) : sqrt(max(T(0), T(1) - Richardson*inv_Pr_turb))^(T(1//4))
-  return buoyancy_factor
+  sqrt(clamp(T(1) - Richardson*inv_Pr_turb, T(0), T(1)))
```
Comment (simonbyrne) [inline src/Atmos/Model/turbulence.jl]: @asraero is this okay?

Comment (asraero) [reply src/Atmos/Model/turbulence.jl]: Yes, `clamp` usage is fine here.

**Pull request merged**